src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/DueDateTypeConfiguratorTest.java

```diff
@@ -1391,4 +1388,85 @@ public void shouldNotRecalculateDateWhenDueDateIsUnconfigurableButOriginDuration
+            .name(CamundaValue.stringValue("priorityDate"))
+            .value(CamundaValue.stringValue(priorityDateValue + "T20:00"))
+            .canReconfigure(CamundaValue.booleanValue(false))
+            .build();
+
+        ConfigurationDmnEvaluationResponse hearingDateIntervalDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("hearingDateIntervalDate"))
+            .value(CamundaValue.stringValue(hearingDateIntervalDateValue + "T20:00"))
+            .canReconfigure(CamundaValue.booleanValue(false))
+            .build();
+
+        ConfigurationDmnEvaluationResponse dueDateOriginRef = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("dueDateOriginRef"))
+            .value(CamundaValue.stringValue("hearingDateIntervalDate"))
+            .canReconfigure(CamundaValue.booleanValue(false))
+            .build();
+
+        assertThatThrownBy(() -> dateTypeConfigurator
+            .configureDates(
+                List.of(nextHearingDate, dueDate, priorityDate, hearingDateIntervalDate, dueDateOriginRef),
+                false,
+                false,
+                taskAttributes
+            ))
+            .isInstanceOf(RuntimeException.class)
+            .hasMessage(INVALID_DATE_REFERENCE_FIELD);
+    }
+
+    @Test
+    public void shouldErrorWhenReferenceDateIsNotYetCalculated() {
+        String priorityDateValue = GIVEN_DATE.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+        String nextHearingDateValue = GIVEN_DATE.plusDays(2).format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+
+        ConfigurationDmnEvaluationResponse nextHearingDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("nextHearingDate"))
+            .value(CamundaValue.stringValue(nextHearingDateValue + "T20:00"))
+            .canReconfigure(CamundaValue.booleanValue(false))
+            .build();
+
+        ConfigurationDmnEvaluationResponse priorityDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("priorityDate"))
+            .value(CamundaValue.stringValue(priorityDateValue + "T20:00"))
+            .canReconfigure(CamundaValue.booleanValue(false))
+            .build();
+
+        ConfigurationDmnEvaluationResponse dueDateOriginRef = ConfigurationDmnEvaluationResponse.builder()
+            .name(CamundaValue.stringValue("dueDateOriginRef"))
+            .value(CamundaValue.stringValue("hearingDateIntervalDate"))
+            .canReconfigure(CamundaValue.booleanValue(false))
+            .build();
+
+        assertThatThrownBy(() -> dateTypeConfigurator
+            .configureDates(
+                List.of(nextHearingDate, priorityDate, dueDateOriginRef),
+                false,
+                false,
+                taskAttributes
+            ))
+            .isInstanceOf(RuntimeException.class)
+            .hasMessage(INVALID_DATE_REFERENCE_FIELD);
+    }
```

Comment:
What about AC1.1 scenario? I.e. when "calculatedDates ordering of mandatory fields is not nextHearingDate, dueDate, priorityDate"

For example "calculatedDates" = "nextHearingDate, priorityDate, dueDate"